aries-site/src/pages/components/textinput.js

```diff
@@ -17,21 +26,144 @@ const TextInput = () => (
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputPasswordExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputPasswordExample />
+        </Example>
+      </Subsection>
+      <Subsection name="With suggestions" level={3}>
+        <SubsectionText>
+          Used to provide the user with available options or suggestions. These
+          suggestions can persist regardless of what the user has entered or can
+          filter based on what the user has typed.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputSuggestionsExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputSuggestionsExample />
+        </Example>
+      </Subsection>
+      <Subsection name="Labeled by icon" level={3}>
+        <SubsectionText>
+          There may be some cases where you need to use a TextInput without a
+          visual label. To meet accessibility requirements, the TextInput can be
+          labeled by another visual context such as an icon with an id.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputLabeledByExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputLabeledByExample />
+        </Example>
+      </Subsection>
+      <Subsection name="Validation" level={3}>
+        <SubsectionText>
+          Used to indicate that a TextInput does not meet the validation
+          requirements of its bounding FormField. Click the Submit button while
+          the TextInput is empty to see the validation state.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputValidationExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputValidationExample />
+        </Example>
+      </Subsection>
+      <Subsection name="Disabled" level={3}>
+        <SubsectionText>
+          Used to indicate that a TextInput cannot be interacted with.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputDisabledExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputDisabledExample />
```

Comment:
FYI, I submitted this PR https://github.com/grommet/grommet/pull/3973 to help get our disabled state closer to the designs.